silver/tests/spec/test_payment_method.py

```diff
@@ -1,49 +1,330 @@
+
+        for old, new in permutations:
+            method = self.create_payment_method(customer=self.customer,
+                                                state=old)
+
+            url = reverse('payment-method-detail', kwargs={
+                'customer_pk': self.customer.pk,
+                'payment_method_id': method.pk
+            })
+            response = self.client.get(url, format='json')
+
+            data = response.data
+            data['state'] = new
+
+            response = self.client.put(url, data=data, format='json')
+
+            message = '{} -> {}: code got {{}} expected {{}}'.format(old, new)
+            if (old, new) in valid_transition_list:
+                self.assertEqual(response.status_code,
+                                 status.HTTP_200_OK,
+                                 message.format(response.status_code,
+                                                status.HTTP_200_OK))
+                self.assertEqual(data, response.data,
+                                 message.format(response.status_code,
+                                                status.HTTP_200_OK))
+            else:
+                self.assertEqual(response.status_code,
+                                 status.HTTP_409_CONFLICT,
+                                 message.format(response.status_code,
+                                                status.HTTP_409_CONFLICT))
+
+    def test_put_detail_additional_data_disabled_state(self):
+        method = self.create_payment_method(customer=self.customer,
+                                            state=PaymentMethod.States.Disabled)
+
+        url = reverse('payment-method-detail', kwargs={
+            'customer_pk': self.customer.pk,
+            'payment_method_id': method.pk
+        })
+        response = self.client.get(url, format='json')
+
+        data = response.data
+        data['additional_data'] = '{"random":"value"}'
+
+        response = self.client.put(url, data=data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_detail_ignore_customer_change(self):
+        other_customer = CustomerFactory.create()
+        method = self.create_payment_method(customer=self.customer)
+
+        url = reverse('payment-method-detail', kwargs={
+            'customer_pk': self.customer.pk,
+            'payment_method_id': method.pk
+        })
+        response = self.client.get(url, format='json')
+
+        expected_data = deepcopy(response.data)
+        data = response.data
+        data['customer'] = reverse('customer-detail',
+                                   request=response.wsgi_request,
+                                   kwargs={'pk': other_customer.pk})
+
+        response = self.client.put(url, data=data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, expected_data)
+
+    @register
```

Comment:
Maybe move this to a common place and maybe rename it to register_processor or something more meaningful.